lms/services/grouping/service.py

```diff
@@ -146,11 +145,59 @@ def get_course_groupings_for_user(
 
         return query.all()
 
+    def get_sections(self, user, lti_user, course, grading_student_id=None):
+        if not self.plugin.sections_type:
+            return None
+
+        if lti_user.is_learner:
+            groupings = self.plugin.get_sections_for_learner(self, course)
+
+        elif grading_student_id:
+            groupings = self.plugin.get_sections_for_grading(
+                self, course, grading_student_id
+            )
+
+        else:
+            groupings = self.plugin.get_sections_for_instructor(self, course)
+
+        return self._to_groupings(user, groupings, course, self.plugin.sections_type)
+
+    # pylint:disable=too-many-arguments
+    def get_groups(self, user, lti_user, course, group_set_id, grading_student_id=None):
+        if not self.plugin.group_type:
+            return None
+
+        if lti_user.is_learner:
+            groupings = self.plugin.get_groups_for_learner(self, course, group_set_id)
+
+        elif grading_student_id:
+            groupings = self.plugin.get_groups_for_grading(
+                self, course, group_set_id, grading_student_id
+            )
+
+        else:
+            groupings = self.plugin.get_groups_for_instructor(
+                self, course, group_set_id
+            )
+
+        return self._to_groupings(user, groupings, course, self.plugin.group_type)
+
+    def _to_groupings(self, user, groupings, course, type_):
```

Comment:
This could become "store_launch_groupings" or similar and add any other relationships here.

If it was public we could call it also for the course grouping.

Reply:
I've mentioned it in slack, but I think a lot of this is kind of cruft as a result of the various APIs sending back different things.

It's expected that Blackboard and Canvas will have different return values, but it's not clear why we don't map them into something we expect before we emit them from the clients on our side.

Ideally these values would require no mapping at all.